metricbeat/mb/testing/modules.go

```diff
@@ -393,6 +393,19 @@ func (r *capturingPushReporterV2) capture(waitEvents int) []mb.Event {
 	}
 }
 
+func (r *capturingPushReporterV2) blockingCapture(waitEvents int) []mb.Event {
+	var events []mb.Event
+	for {
+		select {
+		case e := <-r.eventsC:
```

Comment:
Assuming no events ever arrives, go tests will eventually kill it?

Reply:
I'm not sure if I understood the question, the main pipeline gets aborted after 6 hours (see -> https://github.com/elastic/beats/blob/e6e65aa92fe355c95789691ebf5a3bcecaf5b4ea/Jenkinsfile#L29)

I don't know if the Go test allow some tuning about the timeout per UTs, but if so I don't know how it's configured.

The CI timeout is the very last resort to unfreeze resources in case of any environmental issues, flakiness, deadlocks...  But it could happen that an early stage in the CI gets stalled, so if it's possible to delegate the timeout to the unit-test runner then it will be much faster the feedback.

Reply:
@v1v

the test had a timeout in go of 30 secs, it was tweaked many times but still produces a little flakiness when the job has some resource constraints and the timeout expires.

removing the timeout at all ensure the test is getting enough execution time regardless the CI performances.

@ruflin 
I don't expect to have a case where no events ever arrives, unless we broke something else in the metricset

while 6 hours seem long, I'd prefer to have a stable test that will adapt its time execution to the resource available, that I'm quite confident will never be aborted by the pipeline timeout, rather than setting a "reasonable" timeout in the go test (5 mins?) and deal with the noise of a test that we will end up assuming the flakiness reason for and could hide a real bug.

I'm open to different opinion anyway

Reply:
@ruflin 
I've improved the refactoring in order to avoid a possible race condition in the test (event sent to UDP server before it was actually listening), please give a second review